adapters/sidekiq/lib/opentelemetry/adapters/sidekiq/middlewares/server/tracer_middleware.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module Sidekiq
+      module Middlewares
+        module Server
+          class TracerMiddleware
+            def call(_worker, msg, _queue)
+              parent_context = OpenTelemetry.propagation.extract(msg, extractors: OpenTelemetry.propagation.job_extractors)
```

Comment:
Hey @mwear I was wondering if you could provide a bit of guidance on how to the existing code base to support job extractors and injectors here.  I just kind of shoehorned it in to get a better understanding of the codepath, doesn't seem right though.

I'm still getting familiar with the code base so sorry for the rough state of this PR.

Reply:
:wave: @robertlaurin, sorry for not getting back to you sooner, but work done in #214 makes the answer to this question somewhat easier. Instrumentation adapters should use injectors and extractors registered on the top level propagation object, `OpenTelemetry.propagation`. I did some reorganization to that module in #214. Previously it had `http_injectors` and `http_extractors`. Despite their (previous) name, they will work for many non-http text scenarios. In order to make this as user friendly as possible, I added text injectors / extractors to the propagation object.

With these changes, you should not need special job injectors or extractors. In #214 the top level propagation object looks like this:

```
#inject methods take an optional setter callable
OpenTelemetry.propagation.http.inject(carrier, context)
OpenTelemetry.propagation.text.inject(carrier, context)

# extract methods take an optional getter callable
OpenTelemetry.propagation.http.extract(carrier, context)
OpenTelemetry.propagation.text.extract(carrier, context)
```

For Sidekiq you can use `OpenTelemetry.propagation.text`. We should probably consult @fbogsany to see what he thinks of the changes in #214 though.